wherehows-dao/src/main/java/wherehows/dao/MetadataReadOnlyDao.java

```diff
@@ -0,0 +1,190 @@
+/**
+ * Copyright 2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package wherehows.dao;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.Query;
+import lombok.SneakyThrows;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import wherehows.models.DatasetOwner;
+
+
+/**
+ * Intended to be used for Front-end readonly functions
+ * Use native SQL and then map to entity
+ */
+public class MetadataReadOnlyDao {
```

Comment:
Let's rename this in the next refactoring PR

Reply:
Yes, that's the intention, and the generic functions will be separate into a base class.